### PR TITLE
Fix Grand Calculated Summary Value Source Type

### PR DESCRIPTION
**Pull request by katie-gardner**
### What is the context of this PR?
During development of [GCS Dependencies](https://github.com/ONSdigital/eq-questionnaire-runner/pull/1270) it was noticed that validator doesn't correctly deduce the type for grand calculated summary value sources. This is simply because it was missed the first time and no schema had used a GCS in a when rule against a value. This PR has added the small fix to amend that.

### How to review
Make sure the test schema added (copied from runner) uses the GCS value source as you would expect and that the approach to get the type is working correctly.

### Checklist
* [ ] eq-translations updated to support any new schema keys which need translation

### CODE REVIEW

File: app/validators/routing/types.py [hunk 1]
```diff
@@ -60,6 +60,11 @@ def resolve_value_source_json_type(value_source, schema):
         answer_type = schema.answers_with_context[answer_id]["answer"]["type"]
         return ANSWER_TYPE_TO_JSON_TYPE[answer_type]
 
+    if source == "grand_calculated_summary":
```
Comment (MebinAbraham) [inline app/validators/routing/types.py:63]: A unit test needs adding to `test_types.py`.

Comment (katie-gardner) [reply app/validators/routing/types.py:63]: done

**Review by liamtoozer — APPROVED** [review]
Tested against the PR schema and works great for me 👍

**Review by berroar — APPROVED** [review]

File: app/validators/routing/types.py (hunk 1 above)
Comment (berroar) [reply app/validators/routing/types.py:63]: Resolving on behalf of @MebinAbraham 👍

**Pull request merged**